barretenberg/cpp/src/barretenberg/vm/avm/trace/execution_hints.hpp

```diff
@@ -105,8 +105,15 @@ struct ExecutionHints {
     {
         std::unordered_map<uint32_t, FF> hints_map;
         push_vec_into_map(hints_map, storage_value_hints);
-        push_vec_into_map(hints_map, note_hash_exists_hints);
         push_vec_into_map(hints_map, nullifier_exists_hints);
+        return hints_map;
+    }
+
+    // Leaf index -> exists
+    std::unordered_map<uint32_t, FF> get_leaf_index_hints() const
```

Comment:
Splitting out leaf index hints since they now use the leaf index instead of side effect counters which would likely collide in a single map